crates/network/src/web/client.rs

```diff
@@ -63,77 +67,102 @@ impl ElementComponent for GameClientView {
         hooks.provide_context(|| {
             GameClientRenderTarget(Arc::new(RenderTarget::new(&gpu, uvec2(1, 1), None)))
         });
+
         let (render_target, _) = hooks.consume_context::<GameClientRenderTarget>().unwrap();
 
         let assets = hooks.world.resource(asset_cache()).clone();
-        let game_state = hooks.use_ref_with(|world| {
-            let (systems, resources) = systems_and_resources();
-
-            ClientGameState::new(
-                &gpu,
-                world,
-                assets.clone(),
-                user_id.clone(),
-                render_target.0.clone(),
-                systems,
-                resources,
-            )
-        });
 
         // The game client will be set once a connection establishes
-        let (game_client, set_game_client) = hooks.use_state(None as Option<GameClient>);
+        let (client_state, set_client_state) = hooks.use_state(None as Option<ClientState>);
 
         // When the client is connected, run the update logic each frame
-        if game_client.is_some() {
+        if let Some(client_state) = &client_state {
             tracing::info!("Adding game logic hook");
-            run_game_logic(hooks, game_state.clone(), render_target);
+            run_game_logic(
+                hooks,
+                client_state.game_state.clone(),
+                render_target.clone(),
+            );
         }
 
         // TODO: allow remote shutdown
         let ((_control_tx, control_rx), _) = hooks.use_state_with(|_| flume::unbounded());
 
         let (err, set_error) = hooks.use_state(None);
 
-        hooks.use_local_task(|_| {
+        hooks.use_local_task(move |ui_world| {
+            let local_resources = world_instance_resources(AppResources::from_world(ui_world))
+                .with(game_screen_render_target(), render_target.0.clone());
+
             let task = async move {
+                // NOTE: this is here to simulate a little connection delay to test the
+                // robustness for execution before connect etc
+                //
+                // TODO: remove once the web is fully working
```

Comment:
I suspect this might be the kind of thing we forget about; maybe add a log here to indicate that it slept?

Reply:
Yay, more reasons to add logs.

In all seriousness, I will do that. Makes debugging easier for the web now to test and find things that are running before they should (like the renderer panicking if there is no established connection which I've encountered)